File: library/src/main/java/net/theluckycoder/filechooser/FileArrayAdapter.java
package net.theluckycoder.filechooser;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

class FileArrayAdapter extends ArrayAdapter<Option> {

    private final Context mContext;
    private final int id;
    private final List<Option> items;

    FileArrayAdapter(Context context, int textViewResourceId, List<Option> objects) {
        super(context, textViewResourceId, objects);
        mContext = context;
        id = textViewResourceId;
        items = objects;
    }

    public Option getItem(int i)
    {
        return items.get(i);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        if (convertView == null) {
            LayoutInflater vi = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = vi.inflate(id, null);
        }

        final Option option = items.get(position);

        TextView nameTxt = convertView.findViewById(R.id.name);
        ImageView iconImg = convertView.findViewById(R.id.icon);

        if (option != null) {
            nameTxt.setText(option.getName());
            if (option.isFolder())
                iconImg.setImageResource(R.drawable.ic_folder);
            else
                iconImg.setImageResource(R.drawable.ic_file);

        }

        return convertView;
    }

}

File: library/src/main/java/net/theluckycoder/filechooser/FileArrayAdapter.kt
package net.theluckycoder.filechooser

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

internal class FileArrayAdapter(context: Context, private val id: Int, private val items: List<Option>) : ArrayAdapter<Option>(context, id, items) {

    override fun getItem(i: Int): Option? {
        return items[i]
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var view = convertView
        if (view == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            view = inflater.inflate(id, null)
        }

        val option = items[position]

        val nameTxt = view!!.findViewById<TextView>(R.id.name)
        val iconImg = view.findViewById<ImageView>(R.id.icon)

        nameTxt.text = option.name
        if (option.isFolder)
            iconImg.setImageResource(R.drawable.ic_folder)
        else
            iconImg.setImageResource(R.drawable.ic_file)

        return view
    }

}
